diff -Nur ./1/mapnik-mapnik-91ad6fc/SConstruct ./3/mapnik-mapnik-91ad6fc/SConstruct
--- mapnik-mapnik-91ad6fc/SConstruct	2012-08-23 14:52:52.000000000 +0200
+++ mapnik-mapnik-91ad6fc/SConstruct	2012-10-30 11:05:16.000000000 +0100
@@ -657,6 +657,13 @@
     # does not remove 'lib' and 'include'
     prefixes.insert(0,os.path.dirname(os.path.normpath(env['BOOST_INCLUDES'])))
     prefixes.insert(0,os.path.dirname(os.path.normpath(env['BOOST_LIBS'])))
+    # try to find in custom cflags/ldflags also
+    #for prefix in env['CUSTOM_CXXFLAGS'].split()+env['CUSTOM_LDFLAGS'].split():
+    #    prefix = prefix.replace('-I', '').replace('-L', '')
+    #    fp = os.path.abspath(os.path.dirname(os.path.normpath(prefix)))
+    #    if not fp in prefixes:
+    #        prefixes.insert(0,fp) 
+
     for searchDir in prefixes:
         libItems = glob(os.path.join(searchDir, env['LIBDIR_SCHEMA'], '%s*.*' % search_lib))
         if not libItems:
@@ -1016,9 +1023,15 @@
     env['LIBPATH'] = ['#src']
 
     # set any custom cxxflags and ldflags to come first
-    env.Append(CXXFLAGS = env['CUSTOM_CXXFLAGS'])
-    env.Append(CFLAGS = env['CUSTOM_CFLAGS'])
-    env.Append(LINKFLAGS = env['CUSTOM_LDFLAGS'])
+    #env.Append(CXXFLAGS = env['CUSTOM_CXXFLAGS'])
+    #env.Append(CFLAGS = env['CUSTOM_CFLAGS'])
+    #env.Append(LINKFLAGS = env['CUSTOM_LDFLAGS'])
+    if 'CUSTOM_CXXFLAGS' in env:
+        env.Append(CFLAGS = env['CUSTOM_CXXFLAGS'])
+        env.Append(CXXFLAGS = env['CUSTOM_CXXFLAGS'])
+    # at build time, compilation directories take precedence over user custom cflags/ldflags
+    if 'CUSTOM_LDFLAGS' in env:
+        env.Append(LINKFLAGS = env['CUSTOM_LDFLAGS'])
 
     ### platform specific bits
 
